payment/src/main/java/org/killbill/billing/payment/core/PaymentProcessor.java

```diff
@@ -491,45 +491,66 @@ public boolean apply(final PaymentTransactionModelDao curPaymentTransactionModel
     }
 
     private List<PaymentAttempt> getPaymentAttempts(final List<PaymentTransaction> purchasedTransactions,
-                                                    final List<PaymentAttemptModelDao> paymentAttempts,
+                                                    final List<PaymentAttemptModelDao> pastPaymentAttempts,
                                                     final InternalTenantContext internalTenantContext) {
 
-        List<PaymentAttempt> result = new ArrayList<PaymentAttempt>();
+        List<PaymentAttempt> paymentAttempts = new ArrayList<PaymentAttempt>();
+
+        // Last Attempt from model dao
+        PaymentAttemptModelDao lastPaymentAttemptModelDao = pastPaymentAttempts.get(pastPaymentAttempts.size() - 1);
+        // Last Attempt from Transactions
+        PaymentTransaction lastPaymentTransaction = purchasedTransactions.get(purchasedTransactions.size() - 1);
 
         // Add Past Payment Attempts
-        for (PaymentAttemptModelDao pastPaymentAttempt : paymentAttempts) {
-            DefaultPaymentAttempt paymentAttempt = new DefaultPaymentAttempt(pastPaymentAttempt);
-            result.add(paymentAttempt);
+        for (PaymentAttemptModelDao pastPaymentAttempt : pastPaymentAttempts) {
+            DefaultPaymentAttempt paymentAttempt = new DefaultPaymentAttempt(
+                    pastPaymentAttempt.getAccountId(),
+                    pastPaymentAttempt.getPaymentMethodId(),
+                    pastPaymentAttempt.getId(),
+                    pastPaymentAttempt.getCreatedDate(),
+                    pastPaymentAttempt.getUpdatedDate(),
+                    pastPaymentAttempt.getCreatedDate(),
+                    pastPaymentAttempt.getPaymentExternalKey(),
+                    pastPaymentAttempt.getTransactionId(),
+                    pastPaymentAttempt.getTransactionExternalKey(),
+                    pastPaymentAttempt.getTransactionType(),
+                    pastPaymentAttempt.getStateName(),
+                    pastPaymentAttempt.getAmount(),
+                    pastPaymentAttempt.getCurrency(),
+                    pastPaymentAttempt.getPluginName(),
+                    (lastPaymentTransaction.getPaymentInfoPlugin() != null) ? lastPaymentTransaction.getPaymentInfoPlugin().getProperties() : null);
```

Comment:
Why are you using the `lastPaymentTransaction` to fill in the plugin properties associated with all past attempts? Instead you need to deserialize the `pluginProperties` associated with each attempt.